.github/workflows/build-test.yml

```diff
@@ -491,43 +491,78 @@ jobs:
         # We're not running CI on macOS for now because it's one less matrix entry to lower the number of runners used,
         # macOS runners are expensive, and we assume that Ubuntu is enough to cover the UNIX case.
         os: [ubuntu-latest]
-        python: [3.8] # Use flaky tests to run against more versions of Python.
+        pythonVersion: [3.8]
+        python: [nonConda, conda]
+        jupyter: [raw, local, remote]
```

Comment:
Every scenario is now in the matrix, hence we have all possible scenarios.
I have excluded some with comments, e.g. to reduce unnecessary permutations/scenarios.
Optionally we can add os = windows and double all the tests,
but that might not be an efficient way, we can optionally add `includes` as i have done, this way. I have commented out windows as an example
We can add them later, looks like some stuff aren't working on CI in windows.

Reply:
All of the followign permutations are now generated with test matrix and we have better control & its much simpler to control the scenarios as well